tests/generate_packageTest.py

```diff
@@ -0,0 +1,100 @@
+import copy
+import os
+import shutil
+import sys
+import unittest
+from unittest.mock import patch
+from bears.generate_package import (touch, create_file_from_template,
+                                    create_file_structure_for_packages,
+                                    perform_upload, main,
+                                    create_upload_parser)
+
+
+class touchTest(unittest.TestCase):
+
+    def test_file_doesnt_exist(self):
+        if os.path.exists('TestFile.py'):
+            os.remove('TestFile.py')
+        self.assertFalse(os.path.exists('TestFile.py'))
+        touch('TestFile.py')
+        self.assertTrue(os.path.exists('TestFile.py'))
+
+    def tearDown(self):
+        os.remove('TestFile.py')
+
+
+class create_file_from_templateTest(unittest.TestCase):
+
+    def test_output_file(self):
+        data = {'who': 'George', 'sport': 'swimming'}
+        create_file_from_template(
+            os.path.join(
+                'tests', 'generate_package_input_files', 'template_file.py.in'),
+            os.path.join(
+                'tests', 'generate_package_input_files', 'substituted_file.py'),
+            data)
+        with open(
+                os.path.join(
+                    'tests', 'generate_package_input_files',
+                    'substituted_file.py')) as fl:
+            substituted_file = fl.read()
+        self.assertEqual(substituted_file, 'George has gone swimming again.\n')
+
+    def tearDown(self):
+        os.remove(os.path.join(
+            'tests', 'generate_package_input_files', 'substituted_file.py'))
+
+
+class create_file_structure_for_packagesTest(unittest.TestCase):
+
+    def test_structure(self):
+        touch('TestFile.py')
+        create_file_structure_for_packages('folder', 'TestFile.py', 'Test')
+        self.assertTrue(os.path.exists(
+            os.path.join('folder', 'Test', 'Test', 'Test.py')))
+        self.assertTrue(os.path.exists(
+            os.path.join('folder', 'Test', 'Test', '__init__.py')))
+
+    def tearDown(self):
+        shutil.rmtree('folder')
+
+
+class create_upload_parserTest(unittest.TestCase):
+
+    def test_parser(self):
+        self.assertTrue(create_upload_parser().parse_args(['--upload']).upload)
+
+
+class perform_uploadTest(unittest.TestCase):
+
+    @patch('subprocess.call')
+    def test_command(self, call_mock):
+        perform_upload('.')
+        call_mock.assert_called_with([sys.executable, 'setup.py', 'sdist',
+                                      'bdist_wheel', 'upload', '-r',
+                                      'pypitest'],
+                                     cwd='.')
+
+
+class mainTest(unittest.TestCase):
+
+    def test_main(self):
+        old_argv = copy.deepcopy(sys.argv)
```

Comment:
Why are you copying argv here? I think a comment that explains your approach would help